cfw-builder/steps/enable-ssh-root-login-empty-pw.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+if [ $# -eq 0 ]; then
+    echo "No arguments supplied"
+    exit 1
+fi
+
+ROOT_IMG=$1
+
+SOURCE=/usr/lib/systemd/system/sshd.service
+DEST=/etc/systemd/system/multi-user.target.wants/sshd.service
+
+# symbolic links are not implemented in the e2tools
+# e2ln -s ${ROOT_IMG}:${SOURCE} ${DEST}
+
+# so do a hardlink
+e2ln ${ROOT_IMG}:${SOURCE} ${DEST}
```

Comment:
I've been trying to create my own image using an approach similar to this PR, but struggled to make it work, due to `e2ln` not supporting symlinks. After much digging I finally managed to get `sshd` to start on boot, so I figured I would post here, in case its useful to others.

I discovered that hard links don't seem to work for `systemd`. It appears to only take symlinks in account, which results in `sshd` not starting on boot.

I was able to work around this by using a [systemd generator](https://www.freedesktop.org/software/systemd/man/latest/systemd.generator.html), which runs early in the boot process, and creates the symlink.

I used `e2cp` to create the generator file  at `/usr/lib/systemd/system-generators/sshd-generator` (needs to be executable). The content of the generator file is:

```shell
#!/bin/sh

generator_dir=$1

mkdir -p "$generator_dir/multi-user.target.wants" &&
(
	cd "$generator_dir/multi-user.target.wants" &&
	ln -s /usr/lib/systemd/system/sshd.service .
)
```

This results in SSH starting on boot.